### PR TITLE
feat: bump osx commons contracts

### PR DESCRIPTION
**Pull request by Michael-A-Heuer**
## Description

Use the latest `"@aragon/osx-commons-contracts": "1.4.0-alpha.3",` package

⚠️ Do not spend too much time on reviewing:
- `TokenVoting`, `TokenVotingSetup` 
- `AddresslistVoting`, `AddresslistVotingSetup` 
- `Multisig`, `MultisigSetup` 
- `Admin`, `AdminSetup`

We will simply delete these contracts and all associated helpers and tokens, after we've adapted the deploy script to work without `Multisig`.

Task ID: [OS-1077](https://aragonassociation.atlassian.net/browse/OS-1077)
Task ID: [OS-1219](https://aragonassociation.atlassian.net/browse/OS-1219)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [x] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have updated the `DEPLOYMENT_CHECKLIST` file in the root folder.
- [ ] I have updated the `UPDATE_CHECKLIST` file in the root folder.

Adapting the checklists will be done as part of [OS-924](https://aragonassociation.atlassian.net/browse/OS-924).

[OS-1077]: https://aragonassociation.atlassian.net/browse/OS-1077?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[OS-924]: https://aragonassociation.atlassian.net/browse/OS-924?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[OS-1219]: https://aragonassociation.atlassian.net/browse/OS-1219?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/contracts/test/framework/plugin/plugin-setup-processor.ts [hunk 1]
```diff
@@ -153,15 +154,15 @@ describe('PluginSetupProcessor', function () {
     );
     setupUV2 = await SetupV2.deploy();
 
-    const SetupV3 = await smock.mock<PluginCloneableSetupV2Mock__factory>(
+    const SetupV3 = await smock.mock<PluginUUPSUpgradeableSetupV3Mock__factory>(
```
Comment (Michael-A-Heuer) [inline packages/contracts/test/framework/plugin/plugin-setup-processor.ts:157]: Why was this `PluginCloneableSetupV2Mock__factoryV2Mock`?

File: packages/contracts/package.json [hunk 2]
```diff
@@ -41,7 +41,7 @@
   },
   "homepage": "https://github.com/aragon/osx#readme",
   "dependencies": {
-    "@aragon/osx-commons-contracts": "^1.4.0-alpha.2",
+    "@aragon/osx-commons-contracts": "1.4.0-alpha.3",
```
Comment (Michael-A-Heuer) [inline packages/contracts/package.json:44]: We should always use fixed package versions for contract dependencies.

File: packages/contracts/src/framework/dao/DAOFactory.sol [hunk 3]
```diff
@@ -155,15 +157,21 @@ contract DAOFactory is ERC165, ProtocolVersion {
...
-        dao.initialize(
-            _daoSettings.metadata,
-            address(this),
-            _daoSettings.trustedForwarder,
-            _daoSettings.daoURI
+        // Create a DAO and initialize it. This will give the `ROOT_PERMISSION_`ID` permission to `address(this))`.
+        dao = DAO(
+            payable(
+                daoBase.deployUUPSProxy(
+                    abi.encodeCall(
```
Comment (Michael-A-Heuer) [inline packages/contracts/src/framework/dao/DAOFactory.sol:165]: This immediately initializes the contract and saves us one external call.

File: packages/contracts/src/plugins/governance/admin/AdminSetup.sol
Comment (Michael-A-Heuer) [inline packages/contracts/src/plugins/governance/admin/AdminSetup.sol:1]: Don't review this. This contract will be deleted in https://github.com/aragon/osx/pull/551 soon.

File: packages/contracts/src/plugins/governance/majority-voting/addresslist/AddresslistVotingSetup.sol
Comment (Michael-A-Heuer) [inline packages/contracts/src/plugins/governance/majority-voting/addresslist/AddresslistVotingSetup.sol:1]: Don't review this. This contract will be deleted in https://github.com/aragon/osx/pull/551 soon.

File: packages/contracts/src/plugins/governance/majority-voting/token/TokenVotingSetup.sol
Comment (Michael-A-Heuer) [inline packages/contracts/src/plugins/governance/majority-voting/token/TokenVotingSetup.sol:1]: Don't review this. This contract will be deleted in https://github.com/aragon/osx/pull/551 soon.

File: packages/contracts/src/plugins/governance/multisig/MultisigSetup.sol
Comment (Michael-A-Heuer) [inline packages/contracts/src/plugins/governance/multisig/MultisigSetup.sol:1]: Don't review this. This contract will be deleted in https://github.com/aragon/osx/pull/551 soon.

File: packages/contracts/src/test/plugin/Cloneable/PluginCloneableSetupMock.sol [hunk 4]
```diff
@@ -35,30 +31,40 @@ contract PluginCloneableSetupV1Mock is PluginSetup {
...
-        return address(pluginBase);
+    function prepareInstallation(
+        address _dao,
+        bytes memory
+    ) public virtual override returns (address plugin, PreparedSetupData memory preparedSetupData) {
+        plugin = mockPluginProxy(implementation(), _dao);
+        preparedSetupData.helpers = mockHelpers(1);
+        preparedSetupData.permissions = mockPermissions(5, 6, PermissionLib.Operation.Grant);
     }
-}
```
Comment (jordaniza) [inline packages/contracts/src/test/plugin/Cloneable/PluginCloneableSetupMock.sol:48]: Might be missing something but this looks identical to the mock above, what makes it 'bad'?

File: packages/contracts/test/framework/plugin/plugin-setup-processor.ts [hunk 5]
```diff
@@ -135,49 +139,64 @@ describe('PluginSetupProcessor', function () {
...
+    const implCV1 = await new PluginCloneableV1Mock__factory(
+      signers[0]
+    ).deploy();
     const SetupC1 = await smock.mock<PluginCloneableSetupV1Mock__factory>(
       'PluginCloneableSetupV1Mock'
     );
-    setupCV1 = await SetupC1.deploy();
+    setupCV1 = await SetupC1.deploy(implCV1.address);
 
+    const implCV1Bad = await new PluginCloneableV1MockBad__factory(
```
Comment (jordaniza) [inline packages/contracts/test/framework/plugin/plugin-setup-processor.ts:185]: Again I'm failing to see the need for the 'bad' contract

**Review by jordaniza — APPROVED** [review]
Looks good I just don't get the `MockBad` thing.

**Review by novaknole — APPROVED** [review]

File: packages/contracts/src/test/plugin/Cloneable/PluginCloneableSetupMock.sol (hunk 4 above)
Comment (Michael-A-Heuer) [reply packages/contracts/src/test/plugin/Cloneable/PluginCloneableSetupMock.sol:48]: @jordaniza We have to refactor those in task https://aragonassociation.atlassian.net/browse/OS-1229

File: packages/contracts/test/framework/plugin/plugin-setup-processor.ts (hunk 5 above)
Comment (Michael-A-Heuer) [reply packages/contracts/test/framework/plugin/plugin-setup-processor.ts:185]: @jordaniza We have to refactor those in task https://aragonassociation.atlassian.net/browse/OS-1229

**Pull request merged**